common/reactive/src/main/java/io/helidon/common/reactive/RunnableChain.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c)  2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.common.reactive;
+
+import java.util.ArrayList;
+
+/**
+ * Holds a list of {@link Runnable}s to flatten out a call chain of them.
+ */
+class RunnableChain extends ArrayList<Runnable> implements Runnable {
+    @Override
+    public void run() {
+        for (Runnable inner : this) {
+            inner.run();
+        }
+    }
+
+    public RunnableChain combineWith(Runnable another) {
```

Comment:
Please remove `public` modifier from methods in package local classes

Reply:
fixed